bugbug/bugzilla.py

```diff
@@ -345,47 +345,37 @@ def get_product_component_count(months: int = 12) -> dict[str, int]:
 
 
 def get_active_product_components(products=[]) -> set[tuple[str, str]]:
-    r = utils.get_session("bugzilla").get(
-        "https://bugzilla.mozilla.org/rest/product",
-        params={
-            "type": "accessible",
-            "include_fields": [
-                "name",
-                "is_active",
-                "components.name",
-                "components.is_active",
-            ],
-            "names": products,
-        },
-        headers={"X-Bugzilla-API-Key": Bugzilla.TOKEN, "User-Agent": "bugbug"},
-    )
-    r.raise_for_status()
+    active_components = set()
 
-    return set(
-        (product["name"], component["name"])
-        for product in r.json()["products"]
-        if product["is_active"]
-        for component in product["components"]
-        if component["is_active"]
-    )
+    def product_handler(product):
+        if product["is_active"]:
+            for component in product["components"]:
+                if component["is_active"]:
+                    active_components.add((product["name"], component["name"]))
```

Comment:
As suggested in https://github.com/mozilla/bugbug/pull/3689#discussion_r1348958612, it looks more Pythonic:
```suggestion
            active_components.update(
                (product["name"], component["name"])
                for component in product["components"]
                if component["is_active"]
            )
```